spi/src/main/java/org/opensearch/jobscheduler/spi/utils/LockService.java

```diff
@@ -220,7 +220,7 @@ private void updateLock(final LockModel updateLock, ActionListener<LockModel> li
             listener.onResponse(null);
         } catch (Exception e) {
             logger.error(e);
-            listener.onResponse(null);
+            listener.onFailure(e);
```

Comment:
Should the line at 220 also be handled the same way?

Reply:
This was my initial thought as well, but looking at the usage of `updateLock()` within `renewLock` and `releaseLock` it seems that its meant to return null on response for some cases. 

If stashing the thread context fails for any reason then we'll want to fail that directly.